pallets/pallet-did-lookup/src/lib.rs

```diff
@@ -282,17 +323,46 @@ pub mod pallet {
 		pub fn reclaim_deposit(origin: OriginFor<T>, account: AccountIdOf<T>) -> DispatchResult {
 			let who = ensure_signed(origin)?;
 
-			let record = ConnectedDids::<T>::get(&account).ok_or(Error::<T>::AssociationNotFound)?;
+			let account: LinkableAccountIdOf<T> = account.into();
+			let record = ConnectedDidsV2::<T>::get(&account).ok_or(Error::<T>::AssociationNotFound)?;
 			ensure!(record.deposit.owner == who, Error::<T>::NotAuthorized);
 			Self::remove_association(account)
 		}
+		
+		/// Migrate all associations of a did to the new storage format.
+		#[pallet::weight(<T as Config>::WeightInfo::remove_sender_association())] // @TODO better weight
+		pub fn migrate_associations(origin: OriginFor<T>) -> DispatchResult {
+			let source = <T as Config>::EnsureOrigin::ensure_origin(origin)?;
+
+			// iterate over all linked accounts
+			ConnectedAccounts::<T>::iter_key_prefix(&source.subject()).for_each(|account| {
+
+				// convert account type
+				let linkable_account: LinkableAccountIdOf<T> = account.clone().into();
+
+				// move them into the new storage format
+				ConnectedAccountsV2::<T>::insert(&source.subject(), &linkable_account, ());
+				ConnectedAccounts::<T>::remove(&source.subject(), &account);
+				
+				// retrieve the connection record from the account -> did mapping and move it too
+				if let Some(record) = ConnectedDids::<T>::get(&account) {
+					ConnectedDidsV2::<T>::insert(&linkable_account, &record);
+					ConnectedDids::<T>::remove(&account);
+				}
+			});
+
+			Self::deposit_event(Event::AssociationsMigrated(source.subject()));
+
+			Ok(())
+		}
 	}
 
+
 	impl<T: Config> Pallet<T> {
 		pub(crate) fn add_association(
```

Comment:
What happens if a new account is added to the new storage before the old one is migrated? It would create a situation in which the migration might fail. So we will probably ignore existing keys when migrating over to account for this case, but we should put some comment here.